src/main/java/de/tum/cit/aet/artemis/programming/dto/RepoNameProgrammingStudentParticipationDTO.java

```diff
@@ -0,0 +1,166 @@
+import de.tum.cit.aet.artemis.programming.domain.ProgrammingExerciseTestCaseType;
+import de.tum.cit.aet.artemis.programming.domain.ProgrammingLanguage;
+import de.tum.cit.aet.artemis.programming.domain.ProgrammingSubmission;
+
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public record RepoNameProgrammingStudentParticipationDTO(long id, ZonedDateTime individualDueDate, Set<RepoNameSubmissionDTO> submissions, String participantName,
+        String participantIdentifier, String repositoryUri, String buildPlanId, String branch, RepoNameProgrammingExerciseDTO exercise) {
+
+    /**
+     * Converts a ProgrammingExerciseStudentParticipation into a dto for the endpoint
+     * {@link de.tum.cit.aet.artemis.programming.web.ProgrammingExerciseParticipationResource#getStudentParticipationWithLatestSubmissionLatestResultFeedbacksByRepoName(String)}.
+     *
+     * @param participation to convert
+     * @return the converted DTO
+     */
+    public static RepoNameProgrammingStudentParticipationDTO of(ProgrammingExerciseStudentParticipation participation) {
+        return Optional.ofNullable(participation)
+                .map(p -> new RepoNameProgrammingStudentParticipationDTO(p.getId(), p.getIndividualDueDate(),
+                        Optional.ofNullable(p.getSubmissions()).orElse(Set.of()).stream().filter(s -> s instanceof ProgrammingSubmission)
+                                .map(s -> RepoNameSubmissionDTO.of((ProgrammingSubmission) s)).collect(Collectors.toSet()),
+                        p.getParticipantName(), p.getParticipantIdentifier(), p.getRepositoryUri(), p.getBuildPlanId(), p.getBranch(),
+                        RepoNameProgrammingExerciseDTO.of(p.getProgrammingExercise())))
+                .orElse(null);
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    public record RepoNameSubmissionDTO(long id, Boolean submitted, ZonedDateTime submissionDate, SubmissionType type, Boolean exampleSubmission, Long durationInMinutes,
+            List<RepoNameResultDTO> results, String commitHash, boolean buildFailed) {
+
+        /**
+         * Converts a ProgrammingSubmission into a dto for the endpoint
+         * {@link de.tum.cit.aet.artemis.programming.web.ProgrammingExerciseParticipationResource#getStudentParticipationWithLatestSubmissionLatestResultFeedbacksByRepoName(String)}.
+         *
+         * @param submission to convert
+         * @return the converted DTO
+         */
+        public static RepoNameSubmissionDTO of(ProgrammingSubmission submission) {
+            return Optional.ofNullable(submission)
+                    .map(s -> new RepoNameSubmissionDTO(s.getId(), s.isSubmitted(), s.getSubmissionDate(), s.getType(), s.isExampleSubmission(), s.getDurationInMinutes(),
+                            Optional.ofNullable(s.getResults()).orElse(List.of()).stream().filter(Objects::nonNull).map(RepoNameResultDTO::of).toList(), s.getCommitHash(),
```

Comment:
```suggestion
                            Optional.ofNullable(s.getResults()).orElse(Collections.emptyList()).stream().filter(Objects::nonNull).map(RepoNameResultDTO::of).toList(), s.getCommitHash(),
```
This is a more lightweight since it returns a singleton `EMPTY_LIST` rather than creating a new object every time.

Reply:
sounds good i will incorporate it as soon as possible